neonvm-daemon/cmd/main.go

```diff
@@ -236,3 +273,129 @@ func (s *cpuServer) run(addr string) {
 	}
 	s.logger.Info("CPU server exited without error")
 }
+
+type controlSocketServer struct {
+	// Protects operations from concurrent access. Not sure if this matters for
+	// the internal operations - we could resize swap and set disk quota at the
+	// same time - but seems good to avoid confusion.
+	mutex  *sync.Mutex
+	logger *zap.Logger
+
+	quotaPath      string
+	swapAlreadySet bool
+}
+
+func (s *controlSocketServer) handleResizeSwapOnce(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		s.logger.Error("could not read request body", zap.Error(err))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+
+	size_str := string(body)
+	size_bytes, err := strconv.ParseUint(size_str, 10, 64)
+	if err != nil {
+		s.logger.Error("invalid size in resize-swap-once request", zap.Error(err), zap.String("size_str", size_str))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if s.swapAlreadySet {
+		s.logger.Error("swap was already resized, refusing to resize it again")
+		w.WriteHeader(http.StatusConflict)
+		return
+	}
+
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	s.logger.Info("Resizing swap disk", zap.Uint64("size", size_bytes))
+	err = resizeSwap(s.logger, size_bytes/1024)
+	if err != nil {
+		s.logger.Error("could not resize swap", zap.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	s.swapAlreadySet = true
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *controlSocketServer) handleSetDiskQuota(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		s.logger.Error("could not read request body", zap.Error(err))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+
+	size_str := string(body)
+	size_bytes, err := strconv.ParseUint(size_str, 10, 64)
```

Comment:
I think we should have JSON-based API, for extensibility and multiple arguments.

Should we use https://github.com/OpenAPITools/openapi-generator?

Reply:
+1 for an ad-hoc JSON-based API